src/SecurityManager.ts

```diff
@@ -313,10 +309,8 @@ export async function withSecretStorageKeyCache<T>(func: () => Promise<T>): Prom
     } finally {
         // Clear secret storage key cache now that work is complete
         secretStorageBeingAccessed = false;
-        if (!isCachingAllowed()) {
-            secretStorageKeys = {};
-            secretStorageKeyInfo = {};
-        }
```

Comment:
why does this no longer need to check `secretStorageBeingAccessed`

Reply:
because the line just before this is:

```
        secretStorageBeingAccessed = false;
```

Reply:
Ah yes, so it was redundant all along

Reply:
since https://github.com/matrix-org/matrix-react-sdk/pull/4789, yes